connector/keystone/keystone.go

```diff
@@ -0,0 +1,158 @@
+// Package keystone provides authentication strategy using Keystone.
+package keystone
+
+import (
+	"context"
+	"fmt"
+	"github.com/dexidp/dex/connector"
+	"github.com/sirupsen/logrus"
+	"encoding/json"
+	"net/http"
+	"bytes"
+	"io/ioutil"
+)
+
+var (
+	_ connector.PasswordConnector = &keystoneConnector{}
+  	_ connector.RefreshConnector = &keystoneConnector{}
+)
+
+// Open returns an authentication strategy using Keystone.
+func (c *Config) Open(id string, logger logrus.FieldLogger) (connector.Connector, error) {
+	return &keystoneConnector{c.Domain, c.KeystoneHost,
+	c.KeystoneUsername, c.KeystonePassword, logger}, nil
+}
+
+func (p keystoneConnector) Close() error { return nil }
+
+func (p keystoneConnector) Login(ctx context.Context, s connector.Scopes, username, password string) (
+		identity connector.Identity, validPassword bool, err error) {
+	response, err := p.getTokenResponse(username, password)
+
+	// Providing wrong password or wrong keystone URI throws error
+	if err == nil && response.StatusCode == 201 {
```

Comment:
Let's stick to idiomatic Go code, like
```go
resp, err := p.getTokenResponse(username, password)
if err != nil {
    // handle err
}
if resp.StatusCode == 201 {
    // ... read body
```

Also, `resp.Body` will need to be closed, otherwise we risk leaking memory.